### ISSUE TITLE
Determine options for hosting

### ISSUE THREAD

**Issue by tillig** [autofac/autofac-bot#8]
autofac-bot should be hosted in a location that is owned/accessed by the Autofac organization. Short term, individual ownership is fine, just getting things going; long-term, autofac-bot becomes an issue of 'business continuity' where if the hosting is owned by an individual or if the solution is hosting on a specific person's machine at home, then should that person become unavailable it means the Autofac org is stuck and can't make any changes or updates.

There are some free or cheap cloud-based hosting solutions we may be able to take advantage of. We should research these and figure out the pros and cons of each so we can make a decision.

Documenting what the actual requirements are is probably a good starting point.

**Comment by tillig** [comment]
This ties in with [a conversation from the contributors group.](https://github.com/orgs/autofac/teams/contributors/discussions/27)

**Comment by alsami** [comment]
Has been a long time. I am still interested in working on that. 
While running benchmarks on virtual machine is not optimal, it still might be an option. There are a few low-cost providers that one can choose from that range from 10-20 euros for something very small.

Of course a single benchmark comparison on such a small machine will take much longer to be processed.

What do you think?

**Comment by tillig** [comment]
It's hard to say without actually seeing some sort of comparison. What are the options available to us? How long do benchmarks take on each option?

**Comment by alsami** [comment]
I have some free credits on Azure bc of VS Subscription at work. Will try to setup some virtual machines with similar specs and test against them. Once that is done I'll get back to you with the results.

Are you interested to see a comparison of all available benchmarks or is one specific enough?

**Comment by tillig** [comment]
Going back to the original premise - that whatever solution we have needs to be something owned by the org - it'd be good to come up with a list of options. For example...

- Azure: Could work if a specific subscription is carved out and ownership over that subscription is given to the folks in the org. Probably with its own AAD so we can separately grant and revoke permissions to the subscription.
- AWS or GCP: I don't know what "free" or "$5/year" level sorts of solutions they offer.
- Heroku?
- Something else? Does GitHub have some inherent way to deploy bot sorts of apps?

The question of comparison is more in relation to sort of a matrix of hosting option vs. cost vs. speed. For each option, it'd be more about "pick the fastest cheap-or-free thing." I totally don't know what the options are so I can't say "pick XYZ VM size" or "go with ABC plan."

**Comment by alsami** [comment]
> Going back to the original premise - that whatever solution we have needs to be something owned by the org - it'd be good to come up with a list of options. For example...

Yes, I remember that very well, my idea was just to use Azure VMs temporarily with "similar" specs of VMs to better determine the execution time on VMs.

> Azure: Could work if a specific subscription is carved out and ownership over that subscription is given to the folks in the org. Probably with its own AAD so we can separately grant and revoke permissions to the subscription.

Azure is very costly. There is a free limited subscription for "real" registered organizations but I think that doesn't apply for us.

> AWS or GCP: I don't know what "free" or "$5/year" level sorts of solutions they offer.
> Heroku

Will look that up. Never really used AWS, GCP or Heroku.

> Something else? Does GitHub have some inherent way to deploy bot sorts of apps?

The bot application itself shouldn't really be the problem since it's a simply nodejs application that does just receives requests and passes them along to the ASP.NET Core Application. [Probot itself](https://probot.github.io/docs/deployment/) has some options listed that I'll look into as well.

> The question of comparison is more in relation to sort of a matrix of hosting option vs. cost vs. speed. For each option, it'd be more about "pick the fastest cheap-or-free thing." I totally don't know what the options are so I can't say "pick XYZ VM size" or "go with ABC plan."

Will work something out and get back to you 👍🏽
